# HG changeset patch
# Parent  b0cc0e6c6801ce4593bc7d998d767981e07f9c35
Fix Pacman (Arch Linux) packaging

Like on Debian, hyphens in version numbers aren't supported, so use a
tilde instead.

While we are at it, fix a CMake warning when you have all the
dependencies for the release script but are missing Sphinx to build the
docs.

diff --git a/dist/CMakeLists.txt b/dist/CMakeLists.txt
--- a/dist/CMakeLists.txt
+++ b/dist/CMakeLists.txt
@@ -21,9 +21,11 @@
     pre_release
     package_release
     release_new_tag
-    release_docs
     release_debuild
 )
+IF (SPHINX_FOUND)
+    LIST(APPEND RELEASE_COMMANDS release_docs)
+ENDIF ()
 FOREACH (TARGET ${RELEASE_COMMANDS})
     ADD_CUSTOM_TARGET(
         ${TARGET}
@@ -33,7 +35,9 @@
     )
 ENDFOREACH ()
 
-ADD_DEPENDENCIES(release_docs docs)
+IF (SPHINX_FOUND)
+    ADD_DEPENDENCIES(release_docs docs)
+ENDIF (SPHINX_FOUND)
 
 SET(
     EXTRA_OUTPUT
diff --git a/dist/pkgbuild/.SRCINFO b/dist/pkgbuild/.SRCINFO
--- a/dist/pkgbuild/.SRCINFO
+++ b/dist/pkgbuild/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lightsd
 	pkgdesc = Daemon to control your LIFX smart bulbs via a JSON-RPC API
-	pkgver = {{ version }}
+	pkgver = {{ version|replace("-", "~") }}
 	pkgrel = {{ build_number }}
 	epoch = 1
 	url = https://www.github.com/lopter/lightsd/
diff --git a/dist/pkgbuild/PKGBUILD b/dist/pkgbuild/PKGBUILD
--- a/dist/pkgbuild/PKGBUILD
+++ b/dist/pkgbuild/PKGBUILD
@@ -1,7 +1,9 @@
 # Maintainer: Louis Opter <kalessin@kalessin.fr>
 
 pkgname=lightsd
-pkgver={{ version }}
+pkgver={{ version|replace("-", "~") }}
+# This will have - instead of ~ for rc versions:
+_lightsdver={{ version }}
 pkgrel={{ build_number }}
 epoch=1
 pkgdesc="Daemon to control your LIFX smart bulbs via a JSON-RPC API"
@@ -19,7 +21,7 @@
 install=lightsd.install
 
 build() {
-    cd "$srcdir/$pkgname-$pkgver"
+    cd "$srcdir/$pkgname-$_lightsdver"
 
     cmake .                         \
         -DCMAKE_BUILD_TYPE=RELEASE  \
@@ -30,14 +32,13 @@
 }
 
 check() {
-    cd "$srcdir/$pkgname-$pkgver"
+    cd "$srcdir/$pkgname-$_lightsdver"
 
     make test
 }
 
 package() {
-    cd "$srcdir/$pkgname-$pkgver"
+    cd "$srcdir/$pkgname-$_lightsdver"
 
     make DESTDIR="$pkgdir/" install
 }
-
